### ISSUE TITLE
Product removed from taxon if taxon not shown

### ISSUE THREAD

**Issue by blankcheck** [Sylius/Sylius#9365]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Sylius version   | from 1.0.0-beta2 up

Steps to reproduce (I did it on my older install and on current demo):

1. Go to edit product
2. Have at least one taxon with children, i.e.:

 - taxon 1
 - taxon 2
 - taxon 3
    -- taxon 3.1
    -- taxon 3.2

Add product to taxon 1 and taxon 3.1.
Without expanding taxon 3, add product to taxon 2
As a consequence, product is removed from taxon 3.1

I guess reason being that symfony form doesn't now anything about taxon 3.1 until it was fetched (in current sylius via ajax).

**Comment by blankcheck** [comment]
Just verified that this is bug indeed, here is root cause and initial sub-optimal solution:

In this file https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AdminBundle/Resources/private/js/sylius-lazy-choice-tree.js , function assigned to variable `bindCheckboxAction` basically goes through checkboxes and builds array of **selected** taxons (later that is saved as comma separated string, but not important for bug itself). Of course, taxons which haven't been fetched (because they are 2,3 levels deep) will not be part of selected taxons and will be detached from product. My initial solution to plug the hole:

Override choiceTree method and replace bindCheckboxAction

```
            var bindCheckboxAction = function (checkboxElement) {
                checkboxElement.checkbox({
                    onChange: function () {
                        var $checkboxes = tree.find('.checkbox');
                        var checkedValues = $input.val().split(',');

                        $checkboxes.each(function () {
                            var checkboxValue = $(this).data('value');
                            if ($(this).checkbox('is checked')) {
                                checkedValues.push(checkboxValue);
                            } else {
                                checkedValues = $.grep(checkedValues, function(value) {
                                    return value != checkboxValue;
                                });
                            }
                        });

                        $input.val(checkedValues.join());
                    }
                });
            };


````
This repeats taxons and likely js code should filter against that, but also that fact shouldn't affect taxon assignments as such.

**Comment by pamil** [comment]
Fixed by #9646. 🎉
